eng/pipelines/templates/steps/build-test.yml

```diff
@@ -4,86 +4,76 @@ parameters:
     env:
       GO111MODULE: 'on'
 
-  - pwsh: |
-      $testDirs = (./eng/scripts/get_test_dirs.ps1 -serviceDir $(SCOPE))
-      foreach ($td in $testDirs) {
-        pushd $td
-        Write-Host "##[command]Executing go test -run "^Test" -race -v -coverprofile coverage.txt -covermode atomic $td | go-junit-report -set-exit-code > report.xml"
-        go test -run "^Test" -race -v -coverprofile coverage.txt -covermode atomic . | go-junit-report -set-exit-code > report.xml
-        # if no tests were actually run (e.g. examples) delete the coverage file so it's omitted from the coverage report
-        if (Select-String -path ./report.xml -pattern '<testsuites></testsuites>' -simplematch -quiet) {
-          Write-Host "##[command]Deleting empty coverage file"
-          rm coverage.txt
+  - ${{ if eq(parameters.RunTests, 'true') }}:
+    - pwsh: |
+        go get github.com/jstemmer/go-junit-report
+        go get github.com/axw/gocov/gocov
+        go get github.com/AlekSi/gocov-xml
+        go get github.com/matm/gocov-html
+        go get -u github.com/wadey/gocovmerge
+      displayName: "Install Coverage and Junit Dependencies"
+      workingDirectory: '${{parameters.GoWorkspace}}'
+
+    - pwsh: |
+        $testDirs = (./eng/scripts/get_test_dirs.ps1 -serviceDir $(SCOPE))
+        foreach ($td in $testDirs) {
+          pushd $td
+          Write-Host "##[command]Executing go test -run "^Test" -race -v -coverprofile coverage.txt -covermode atomic $td | go-junit-report -set-exit-code > report.xml"
+          go test -run "^Test" -race -v -coverprofile coverage.txt -covermode atomic . | go-junit-report -set-exit-code > report.xml
+          # if no tests were actually run (e.g. examples) delete the coverage file so it's omitted from the coverage report
+          if (Select-String -path ./report.xml -pattern '<testsuites></testsuites>' -simplematch -quiet) {
+            Write-Host "##[command]Deleting empty coverage file"
+            rm coverage.txt
+          }
         }
-      }
-    displayName: 'Run Tests'
-    workingDirectory: '${{parameters.GoWorkspace}}'
-    env:
-      GO111MODULE: 'on'
+      displayName: 'Run Tests'
+      workingDirectory: '${{parameters.GoWorkspace}}'
+      env:
+        GO111MODULE: 'on'
 
-  - pwsh: |
-      $coverageFiles = [Collections.Generic.List[String]]@()
-      Get-Childitem -recurse -path $(SCOPE) -filter coverage.txt | foreach-object {
-        $covFile = $_.FullName
-        Write-Host "Adding $covFile to the list of code coverage files"
-        $coverageFiles.Add($covFile)
-      }
-      gocovmerge $coverageFiles > mergedCoverage.txt
-      gocov convert ./mergedCoverage.txt > ./coverage.json
-      # gocov converts rely on standard input
-      Get-Content ./coverage.json | gocov-xml > ./coverage.xml
-      Get-Content ./coverage.json | gocov-html > ./coverage.html
-    displayName: 'Generate Coverage XML'
-    workingDirectory: '${{parameters.GoWorkspace}}sdk'
+    - pwsh: |
```

Comment:
Perhaps in a separate PR to keep the diffs small, but it would be nice to move these three powershell blocks into a script/scripts themselves to be consistent with build.ps1 and keep the yaml small (and allow devs to easily run the same test code as the pipeline).

Reply:
Fully agree here.

Reply:
https://github.com/Azure/azure-sdk-for-go/issues/15113